lightly/loss/barlow_twins_loss.py

```diff
@@ -0,0 +1,47 @@
+import torch
+
+class BarlowTwinsLoss(torch.nn.Module):
+    """Implementation of the Barlow Twins Loss from Barlow Twins[0] paper.
+    This code specifically implements the Figure Algorithm 1 from [0].
+    [0] Zbontar,J. et.al, 2021, Barlow Twins... https://arxiv.org/abs/2103.03230
+
+        Examples:
+
+        >>> # initialize loss function
+        >>> loss_fn = BarlowTwinsLoss()
+        >>>
+        >>> # generate two random transforms of images
+        >>> t0 = transforms(images)
+        >>> t1 = transforms(images)
+        >>>
+        >>> # feed through SimSiam model
+        >>> out0, out1 = model(t0, t1)
+        >>>
+        >>> # calculate loss
+        >>> loss = loss_fn(out0, out1)
+
+    """
+
+    def __init__(self, device, lambda_param=5e-3):
```

Comment:
I would not require the device here but rather get the device during the forward pass as you can see in our [`NTXEntLoss` (this is a link)](https://github.com/lightly-ai/lightly/blob/develop/lightly/loss/ntx_ent_loss.py) on line 128.

In your `forward` implementation it could be something like:
```Python

device = z_a.device
```

Reply:
Perfect! I'll commit soon

Reply:
Done